pkg/controller/direct/secretmanager/secret_controller.go

```diff
@@ -133,6 +128,7 @@ type Adapter struct {
 	gcpClient *gcp.Client
 	desired   *krm.SecretManagerSecret
 	actual    *secretmanagerpb.Secret
+	u         *unstructured.Unstructured
```

Comment:
I do think if we stored the desired state as proto this might get simpler ... probably a separate PR, though it might be easier than adding `u` to the Adapter

Reply:
Sg, I can do that in a follow-up PR.